src/nxt_conf_validation.c

```diff
@@ -1250,107 +1358,235 @@ nxt_conf_vldt_array_iterator(nxt_conf_validation_t *vldt,
     }
 }
 
-
 static nxt_int_t
-nxt_conf_vldt_system(nxt_conf_validation_t *vldt, nxt_conf_value_t *value,
-    void *data)
+nxt_conf_vldt_environment(nxt_conf_validation_t *vldt, nxt_str_t *name,
+    nxt_conf_value_t *value)
 {
-    size_t                  length;
-    nxt_str_t               name;
-    nxt_conf_vldt_system_t  validator;
-    char                    string[32];
+    nxt_str_t  str;
 
-    /* The cast is required by Sun C. */
-    validator = (nxt_conf_vldt_system_t) data;
+    if (name->length == 0) {
+        return nxt_conf_vldt_error(vldt,
+                                   "The environment name must not be empty.");
+    }
 
-    nxt_conf_get_string(value, &name);
+    if (nxt_memchr(name->start, '\0', name->length) != NULL) {
+        return nxt_conf_vldt_error(vldt, "The environment name must not "
+                                   "contain null character.");
+    }
+
+    if (nxt_memchr(name->start, '=', name->length) != NULL) {
+        return nxt_conf_vldt_error(vldt, "The environment name must not "
+                                   "contain '=' character.");
+    }
 
-    length = name.length + 1;
-    length = nxt_min(length, sizeof(string));
+    if (nxt_conf_type(value) != NXT_CONF_STRING) {
+        return nxt_conf_vldt_error(vldt, "The \"%V\" environment value must be "
+                                   "a string.", name);
+    }
 
-    nxt_cpystrn((u_char *) string, name.start, length);
+    nxt_conf_get_string(value, &str);
 
-    return validator(vldt, string);
+    if (nxt_memchr(str.start, '\0', str.length) != NULL) {
+        return nxt_conf_vldt_error(vldt, "The \"%V\" environment value must "
+                                   "not contain null character.", name);
+    }
+
+    return NXT_OK;
 }
 
+typedef struct {
+    u_char usr;
+    u_char mnt;
+    u_char net;
+    u_char ipc;
+    u_char uts;
+    u_char pid;
+    u_char cgr;
+} nxt_conf_vldt_clone_namespaces_conf_t;
```

Comment:
Most of these fields are unused. Only `ipc` is checked.
On the other hand, `nxt_init_set_ns()` manually iterate over object instead of using such kind of mapping.